packages/grid/_modules_/grid/hooks/features/dimensions/useGridDimensions.ts

```diff
@@ -88,18 +89,15 @@ export function useGridDimensions(
       rootElement.removeChild(scrollDiv);
     }
 
-    const currentPage = getCurrentPageRows(apiRef.current.state, {
-      pagination: props.pagination,
-      paginationMode: props.paginationMode,
-    });
-    const pageScrollHeight = currentPage.rows.length * rowHeight;
     const viewportOuterSize: ElementSize = {
       width: rootDimensionsRef.current.width,
-      height: props.autoHeight ? pageScrollHeight : rootDimensionsRef.current.height - headerHeight,
+      height: props.autoHeight
+        ? rowsMeta.totalHeight
```

Comment:
At 1st I thought it was the sum for all rows

```suggestion
        ? rowsMeta.currentPageTotalHeight
```